clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ClusterCachingAgent.groovy

```diff
@@ -550,7 +550,9 @@ class ClusterCachingAgent implements CachingAgent, OnDemandAgent, AccountAware,
           cacheTargetGroups(data, targetGroups)
           cacheLaunchTemplate(data, launchTemplates)
         } catch (Exception ex) {
-          log.warn("Failed to cache ${asg.autoScalingGroupName} in ${account.name}/${region}", ex)
```

Comment:
I'm surprised this log statement doesn't print a stack trace already.

Reply:
Was going to comment the same but figured I must be missing something obvious.

Reply:
Here's the log it provides, unfortunately: ```message: Failed to cache METRICS-C-UW2-sb-Metrics Synthetics_asg in jnkscpn-c-uw2/us-west-2
   stack_trace: java.lang.NullPointerException: null```

Reply:
For what it's worth, we find lots of other useful stack traces in other locations in the logs. Not sure why this one is special, but it very much is.

Reply:
Searching a bit through the code, I see that log.warn('stuff', exception) is uncommon in the code. https://github.com/spinnaker/clouddriver/blob/master/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/http/HttpArtifactConfiguration.java#L53 https://github.com/spinnaker/clouddriver/blob/master/cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cache/SqlCache.kt#L917 https://github.com/spinnaker/clouddriver/blob/master/clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/health/AzureHealthIndicator.groovy#L79 https://github.com/spinnaker/clouddriver/blob/master/clouddriver-web/src/main/java/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java#L64...

Reply:
This might be explained if http://www.slf4j.org/api/org/slf4j/Logger.html somehow it was resolving the function call to log.warn(format, object) instead of log.warn(msg, Throwable)

Reply:
https://github.com/spinnaker/kork/blob/447d0fb0df1550df2d961ffa8f8e80137ea73e4f/kork-web/src/main/java/com/netflix/spinnaker/kork/web/exceptions/GenericExceptionHandlers.java#L146 is one place that logs a stack trace that happens relatively often.

Reply:
Would you mind pasting the stack trace you see with this code change?  Maybe it'll show a different place to fix such that this log statement starts to work properly?

Reply:
Wonder if the fact that this is groovy is causing grief here?

Reply:
Erm, well - I would have to deploy it onto our local Spinnaker servers to be able to answer that, and I don't have access to do that.

Reply:
I presume we're on a later slf4j than 1.6.0? This link smells like what we're seeing: https://stackoverflow.com/questions/5951209/how-to-log-exception-and-message-with-placeholders-with-slf4j

Reply:
Let me put together a quick sample app to tinker with this...

Reply:
FYI:
https://github.com/spinnaker/spinnaker/issues/5473
i think is also relevant ;) I'm watching this & other thread ;) Was rather confused to why stack traces aren't working myself.